dvc/remote/ssh/connection.py

```diff
@@ -320,3 +329,31 @@ def md5(self, path):
     def cp(self, src, dest):
         self.makedirs(posixpath.dirname(dest))
         self.execute("cp {} {}".format(src, dest))
+
+    def open_sftp(self):
+        return self._ssh.open_sftp()
+
+    def open_max_sftp_channels(self):
+        channels = []
+
+        with self.mute_paramiko():
+            while True:
+                try:
+                    channels.append(self.open_sftp())
+                except paramiko.ssh_exception.ChannelException:
+                    if not channels:
+                        raise
+                    return channels
+                else:
+                    continue
+
+    @contextmanager
+    def mute_paramiko(self):
+        paramiko_logger = logging.getLogger("paramiko")
+        old_level = paramiko_logger.getEffectiveLevel()
+        paramiko_logger.setLevel(logging.CRITICAL)
+
+        try:
+            yield
+        finally:
+            paramiko_logger.setLevel(old_level)
```

Comment:
does it make sense to set the CRITICAL level globally? do we care about the logs from paramiko at all?

Reply:
If we don't do this, it will log something like: `Secsh channel open FAILED: open failed: Administratively prohibited` when trying to figure out the max possible connections

Reply:
yes, so my question is - do we need to see their logs at all? Can we set it to critical globally?

Reply:
I'm not sure if it would be a good idea, if something goes wrong, then we will be clueless

Reply:
but making it silent here we are already going that path, right? if we think there is some value paramiko can provide in terms of logging, why don't we switch it on when `-v` is passed?

Reply:
agree, @shcheklein , I'll add support for `-v`